nixos/lib/test-driver/test-driver.py

```diff
@@ -566,6 +567,42 @@ def screenshot(self, filename: str) -> None:
             if ret.returncode != 0:
                 raise Exception("Cannot convert screenshot")
 
+    def copy_from_host_via_shell(self, source: str, target: str) -> None:
+        """Copy a file from the host into the guest by piping it over the
+        shell into the destination file. Works without host-guest shared folder.
+        Prefer copy_from_host for large files.
```

Comment:
```suggestion
        Prefer copy_from_host whenever possible.
```

This implies it's okay to use it for small files.

In fact, we should use `copy_from_host` whenever possible, and only use the base64 and shell dance if we have to IMHO.